src/status_im/contexts/wallet/wallet_connect/effects.cljs

```diff
@@ -145,3 +145,35 @@
            :reason      reason})
          (promesa/then on-success)
          (promesa/catch on-error)))))
+
+(rf/reg-fx
+ :effects.wallet-connect/should-session-request-handled
+ (fn [{:keys [db event on-handle on-do-not-handle]}]
+   (let [{:keys [topic]} event
+         should-handle?  (some #(= topic %)
+                               (map :topic (:wallet-connect/sessions db)))]
+     (if should-handle?
+       (on-handle)
+       (do
+         (rf/dispatch [:wallet-connect/send-response
+                       {:error (wallet-connect/get-sdk-error
+                                constants/wallet-connect-user-rejected-chains-error-key)}])
+         (on-do-not-handle))))))
+
+;; (rf/reg-event-fx
+;;  :wallet-connect/on-session-delete
+;;  (fn [{:keys [db]} [{:keys [topic] :as event}]]
+;;    (log/info "Received Wallet Connect session delete: " event)
+;;    {:fx [[:effects.wallet-connect/should-session-request-be-handled?
+;;           {:db        db
+;;            :event     event
+;;            :on-handle #(rf/dispatch [:wallet-connect/disconnect-session topic])}]]}))
+
+;; (rf/reg-event-fx
+;;  :wallet-connect/on-session-request
+;;  (fn [{:keys [db]} [event]]
+;;    {:fx [[:effects.wallet-connect/should-session-request-be-handled?
+;;           {:db        db
+;;            :event     event
+;;            :on-handle #(rf/dispatch
+;;                         [:wallet-connect/process-session-request event])}]]}))
```

Comment:
is this commented out bc it's an example of where it will be used?

Reply:
This is not an example, these events are already defined (in a slightly different way) in Alwx's PR. I didn't want to change that yet because there will be merge-conflicts. 

When his code gets merged, all I need to do is replace his definitions with the ones above.